oem/ibm/libpldmresponder/file_io_type_lid.hpp

```diff
@@ -278,12 +304,39 @@ class LidHandler : public FileHandler
                     oemIbmPlatformHandler = dynamic_cast<
                         pldm::responder::oem_ibm_platform::Handler*>(
                         oemPlatformHandler);
+                // validate api
+                auto& bus = pldm::utils::DBusHandler::getBus();
                 auto sensorId =
                     oemIbmPlatformHandler->codeUpdate->getMarkerLidSensor();
+                auto markerLidName = "80a00001.lid";
+                auto markerLidDirPath =
+                    fs::path(LID_STAGING_DIR) / "lid" / markerLidName;
                 using namespace pldm::responder::oem_ibm_platform;
+                try
+                {
+                    rc = processCodeUpdateLid(lidPath);
+                    if (rc != PLDM_SUCCESS)
+                    {
+                        return rc;
+                    }
+                    auto method = bus.new_method_call(
+                        "xyz.openbmc_project.Software.BMC.Updater",
+                        "/xyz/openbmc_project/software",
+                        "xyz.openbmc_project.Software.LID", "Validate");
+                    method.append(markerLidDirPath.c_str());
+                    bus.call_noreply(method);
+                }
+                catch (const sdbusplus::exception::exception& e)
+                {
+                    std::cerr << "Marker lid validate error, "
+                              << "ERROR=" << e.what() << std::endl;
+                    oemIbmPlatformHandler->sendStateSensorEvent(
+                        sensorId, PLDM_STATE_SENSOR_STATE, 0, ENTITLEMENT_FAIL,
```

Comment:
Is this event handled by host? Did we test this?

Reply:
Yes it's handled on the host side, Yes I have verified the behaviour. So PFW is responsible for the marker lid validate and it fails the update if we have sent this event through PHYP to PFW.